src/favorites.rs

```diff
@@ -35,14 +35,14 @@ pub(crate) fn list_favorites(args: &Args) -> Result<()> {
 }
 
 pub(crate) fn resolve_favorite(args: &mut Args) -> Result<()> {
-    if args.git.is_some() {
+    if args.git.is_some() || args.folder.is_some() {
         return Ok(());
     }
 
     let favorite_name = args
         .favorite
         .as_ref()
-        .ok_or_else(|| anyhow!("Please specify either --git option, or a predefined favorite"))?;
+        .with_context(|| "Please specify either --git, --folder, or a predefined favorite")?;
```

Comment:
`.with_context` is only needed when you want to format the message with some variables where you would normally use `format!` but then use `.with_context` instead. (reason: the callback is only evaluated when the error appears, so the formatting of the string has only costs when the error case happens)

But here you don't format anything, so what you would use in this case is simply only `.context("Please specify either --git, --folder, or a predefined favorite")` As you can see there is no closure argument, just the plain message.

Reply:
Accepted 😄